### ISSUE TITLE
ClientRegressionWithMockNetworkTest.testNearCache_WhenRegisteredNodeIsDead

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#7948]
```
java.lang.AssertionError: expected:<null> but was:<b>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest$7.run(ClientRegressionWithMockNetworkTest.java:481)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:851)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:865)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testNearCache_WhenRegisteredNodeIsDead(ClientRegressionWithMockNetworkTest.java:478)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-OracleJDK1.6/com.hazelcast$hazelcast-client/368/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testNearCache_WhenRegisteredNodeIsDead/

**Comment by cangencer** [comment]
Looking at this more in detail, this test should fail if listener for near cache is not tried to re-register.

**Comment by cangencer** [comment]
Not reproducible after running for 1h both on local and EC2, and no other recent test failures. I think this was a regression, fixed by #7944
